include/vast/Conversion/TypeConverters/TypeConverter.hpp

```diff
@@ -139,6 +152,10 @@ namespace vast::conv::tc {
                 .template take_wrapped< maybe_type_t >();
         }
 
+        maybe_type_t convert_arg_type(mlir_type t, unsigned long /* idx */) const {
```

Comment:
I see that `convert_arg_type` here takes an additional parameter `idx` which is currently unused, but forces us to iterate over arguments using `llvm::enumerate` later on. Are we sure we'll need the `idx` parameter in the future?

Reply:
It is there as an extension point, there are overloads that use the index.